src/AzureExtension/Helpers/IconLoader.cs

```diff
@@ -12,7 +12,7 @@ public class IconLoader
     public static string GetIconAsBase64(string filename)
     {
         Log.Logger()?.ReportDebug(nameof(IconLoader), $"Asking for icon: {filename}");
-        if (!Base64ImageRegistry.ContainsKey(filename))
+        if (!Base64ImageRegistry.TryGetValue(filename, out var _))

```

Comment:
You can use TryAdd() here instead and then remove the separate add command inside the conditional.

Reply:
Changed.